### PR TITLE
#172 Support for more image options

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-design/issues/172 |
| Dependencies | -- |
| Decisions | Support more image options. (Saw which ones were missing from here https://github.com/ripe-tech/ripe-compose/blob/master/src/ripe_compose/controllers/web/composition.py#L150-L210) |
| Animated GIF | -- |

### CODE REVIEW

**Comment by gcandal** [comment]
@NFSS10 before accepting this, please double-check that everything is aligned between:
- This PR
- https://docs.platforme.com/#render-endpoints-compose
- https://github.com/ripe-tech/plugin-base/blob/master/templates/config/applier.html.tpl
- https://github.com/ripe-tech/ripe-core/blob/de5eb20f0ed326921217efdc422cd08cffb9094b/src/ripe_core/controllers/api/base.py#L404
- https://github.com/ripe-tech/ripe-compose/blob/874a102570643a17c606ba67a8d152c4b72ec53a/src/ripe_compose/controllers/web/composition.py#L32
- https://github.com/ripe-tech/ripe-compose/blob/51f1eb5c5ea141d7fa98302fadd6d0b85cefffd2/src/ripe_compose/util/interface.py
- https://github.com/ripe-tech/ripe-compose/blob/51f1eb5c5ea141d7fa98302fadd6d0b85cefffd2/src/ripe_compose/util/configurator.py#L2521
- other related ripe-compose stuff

This seemingly repetitive work is **key** to prevent future bugs that will be hell to diagnose, so take the time to do this properly.

**Comment by NFSS10** [comment]
> 
> 
> @NFSS10 before accepting this, please double-check that everything is aligned between:
> 
>     * This PR
> 
>     * https://docs.platforme.com/#render-endpoints-compose
> 
>     * https://github.com/ripe-tech/plugin-base/blob/master/templates/config/applier.html.tpl
> 
>     * https://github.com/ripe-tech/ripe-core/blob/de5eb20f0ed326921217efdc422cd08cffb9094b/src/ripe_core/controllers/api/base.py#L404
> 
>     * https://github.com/ripe-tech/ripe-compose/blob/874a102570643a17c606ba67a8d152c4b72ec53a/src/ripe_compose/controllers/web/composition.py#L32
> 
>     * https://github.com/ripe-tech/ripe-compose/blob/51f1eb5c5ea141d7fa98302fadd6d0b85cefffd2/src/ripe_compose/util/interface.py
> 
>     * https://github.com/ripe-tech/ripe-compose/blob/51f1eb5c5ea141d7fa98302fadd6d0b85cefffd2/src/ripe_compose/util/configurator.py#L2521
> 
>     * other related ripe-compose stuff
> 
> 
> This seemingly repetitive work is **key** to prevent future bugs that will be hell to diagnose, so take the time to do this properly.

I think it is good, but as we discussed, there are missing stuff for example in docs.platforme.
I will update those too.

File: src/js/visual/image.js [hunk 1]
```diff
@@ -162,7 +308,53 @@ ripe.Image.prototype.update = async function(state, options = {}) {
         height: height,
         crop: crop,
         initials: initialsSpec.initials,
-        profile: initialsSpec.profile
+        profile: initialsSpec.profile,
+        rotation: this.rotation,
```
Comment (gcandal) [inline src/js/visual/image.js:312]: Make this follow the same order as in the files above.

Comment (gcandal) [reply src/js/visual/image.js:312]: (above `crop`)

**Review by gcandal — CHANGES_REQUESTED** [review]

**Review by gcandal — CHANGES_REQUESTED** [review]
I've just noticed there's an important naming standard that must be applied here:
- Naming in GET query params: `snake_case`
- Naming in Javascript variables: `camelCase`

This means that lines must be `params.bounding_box = options.boundingBox`, so that I can use it like:

```
ripe.bindImage(..., boundingBox: ...)
```
```

**Comment by NFSS10** [comment]
I double checked the names, I think they are all correct, but It's probably better if you do a quick look @gcandal

**Review by gcandal — APPROVED** [review]

File: src/js/base/api.js [hunk 2]
```diff
@@ -651,6 +651,190 @@ ripe.Ripe.prototype._getImageOptions = function(options = {}) {
         params.initials = initials;
     }
 
+    if (options.profile !== undefined && options.profile !== null) {
```
Comment (joamag) [inline src/js/base/api.js:654]: we already have this one... what's the catch ?

**Review by joamag — CHANGES_REQUESTED** [review]

File: src/js/base/api.js (hunk 2 above)
Comment (gcandal) [reply src/js/base/api.js:654]: There's none, I've missed and I think @NFSS10 didn't see the `params.initials_profile = options.profile.join(",");` above. Right?

Comment (NFSS10) [reply src/js/base/api.js:654]: Yes. it was that

Comment (NFSS10) [reply src/js/base/api.js:654]: Yup, I think that was what happened

Comment (gcandal) [reply src/js/base/api.js:654]: I guess you can delete `options.initialsProfile` and `options.initialsProfiles` as well, given that `options.profile` is already a list that covers every case.

File: src/js/base/api.js [hunk 3]
```diff
@@ -651,6 +651,186 @@ ripe.Ripe.prototype._getImageOptions = function(options = {}) {
...
+        params.engine = options.engine;
+    }
+
+    if (options.initialsProfile !== undefined && options.initialsProfile !== null) {
+        params.initials_profile = options.initialsProfile;
+    }
+
+    if (options.initialsProfiles !== undefined && options.initialsProfile !== null) {
+        params.initials_profiles = options.initialsProfiles;
+    }
```
Comment (gcandal) [inline src/js/base/api.js:688]: Let's just keep the existing `options.profile`

File: src/js/visual/image.js [hunk 4]
```diff
@@ -160,9 +289,53 @@ ripe.Image.prototype.update = async function(state, options = {}) {
         size: size,
         width: width,
         height: height,
+        rotation: this.rotation,
         crop: crop,
         initials: initialsSpec.initials,
-        profile: initialsSpec.profile
+        profile: initialsSpec.profile,
+        flip: this.flip,
```
Comment (gcandal) [inline src/js/visual/image.js:296]: `flip: flip` so that it is aligned with how `crop` works.

**Review by gcandal — CHANGES_REQUESTED** [review]

File: src/js/visual/image.js (hunk 4 above)
Comment (NFSS10) [reply src/js/visual/image.js:296]: Not sure if I understood

File: src/js/visual/image.js [hunk 5]
```diff
@@ -53,8 +53,52 @@ ripe.Image.prototype.init = function() {
...
+    this.initialsBlendColor = this.options.initialsBlendColor || null;
+    this.initialsPattern = this.options.initialsPattern || null;
+    this.initialsTexture = this.options.initialsTexture || null;
+    this.initialsExclusion = this.options.initialsExclusion || null;
+    this.initialsInclusion = this.options.initialsInclusion || null;
+    this.initialsImageRotation = this.options.initialsImageRotation || null;
+    this.initialsImageFlip = this.options.initialsImageFlip || null;
+    this.initialsImageMirror = this.options.initialsImageMirror || null;
+    this.debug = this.options.debug || null;
+    this.initialsDebug = this.debug ? this.debug : this.options.initialsDebug || null;
```
Comment (gcandal) [inline src/js/visual/image.js:87]: Let's get rid of alias

**Review by gcandal — CHANGES_REQUESTED** [review]

**Review by gcandal — APPROVED** [review]

**Review by joamag — APPROVED** [review]

**Comment by joamag** [comment]
@NFSS10 let's create a simple unit test to assert the generation of the URL

**Pull request merged**